### PR TITLE
BACK-1378: Restrict SwaapV2 on pool level

### PR DESCRIPTION
**Pull request by JackieJoo**

### CODE REVIEW

File: src/dex/swaap-v2/swaap-v2.ts [hunk 1]
```diff
@@ -23,6 +29,8 @@ import {
   SwaapV2APIParameters,
   SwaapV2QuoteError,
   TokensMap,
+  SwaapV2NotificationRequest,
```
Comment (KanievskyiDanylo) [inline src/dex/swaap-v2/swaap-v2.ts:32]: Unused variable

**Review by KanievskyiDanylo — APPROVED** [review]

File: src/dex/swaap-v2/swaap-v2.ts (hunk 1 above)
Comment (JackieJoo) [reply src/dex/swaap-v2/swaap-v2.ts]: fixed

File: src/dex/swaap-v2/swaap-v2.ts [hunk 2]
```diff
@@ -578,33 +593,62 @@ export class SwaapV2 extends SimpleExchange implements IDex<SwaapV2Data> {
...
+    await this.dexHelper.cache.hset(
+      this.runtimeMMsRestrictHashMapKey,
+      poolIdentifier,
+      Date.now().toString(),
     );
 
-    return result === 'true';
+    this.rateFetcher
+      .notify(SWAAP_BANNED_CODE, message, this.getNotifyReqParams())
+      .then((answer: SwaapV2NotificationResponse) => {
```
Comment (aburkut) [inline src/dex/swaap-v2/swaap-v2.ts:627]: @JackieJoo Could we use async/await ?

Comment (JackieJoo) [reply src/dex/swaap-v2/swaap-v2.ts:627]: we'd need to wrap it in a separate try-catch, seems easier to just use `.then().catch()`

**Review by aburkut — APPROVED** [review]

**Pull request merged**